File: TaskManagementSystemCleanArchitecture/Application/DTOs/TaskResponseDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Application.DTOs
{
    public class TaskResponseDto
    {
        public int TaskId { get; set; }
        public string Title { get; set; } = null!;
        public string? Description { get; set; }
        public string Status { get; set; } = null!; 
        public string CreatedByUserName { get; set; } = null!; 
        public DateTime CreatedDate { get; set; }
        public DateTime? DueDate { get; set; }

        // قائمة باليوزرز اللي شغالين على التاسك وتفاصيلهم فيها
        public List<AssignedUserDto> AssignedUsers { get; set; } = new();
    }
}

File: Frontend/TaskSystem/src/app/shared/models/task.response.dto.ts
import { AssignedUserDto } from "./assigned-task-dto";

export interface TaskResponseDto {
  taskId: number;
  title: string;
  description?: string;
  status: string; 
  createdByUserName: string; 

   // match enum backend
  createdDate: Date;      // Date
  dueDate?: Date;
  closedDate?: Date;
  
  assignedUsers: AssignedUserDto[];

}